File: zig/difference-of-squares/difference_of_squares.zig
pub fn squareOfSum(number: isize) isize {
    var result: isize = 0;
    var n: isize = 0;
    while (n <= number) : (n += 1) {
        result += n;
    }

    return result * result;
}

pub fn sumOfSquares(number: isize) isize {
    var result: isize = 0;
    var n: isize = 0;
    while (n <= number) : (n += 1) {
        result += n * n;
    }

    return result;
}

pub fn differenceOfSquares(number: isize) isize {
    const squared_sum = squareOfSum(number);
    const sum_squares = sumOfSquares(number);

    return @max(sum_squares - squared_sum, squared_sum - sum_squares);
}

File: ocaml/difference-of-squares/difference_of_squares.ml
let rec accumulate acc v func =
    if acc = 0 then v
    else accumulate (acc - 1) (func v acc) func

let square_of_sum n = 
    let sum_up acc v = acc + v in
    let s = accumulate n 0 sum_up in
    s * s

let sum_of_squares n =
    let square_and_add acc v = acc + (v * v) in
    accumulate n 0 square_and_add

let difference_of_squares n =
    square_of_sum n - sum_of_squares n
